src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,544 @@
+        /// </summary>
+        public bool InvertedColors
+        {
+            get => _invd;
+            set
+            {
+                _invd = value;
+                SpiWrite(false, _invd ? new byte[] { (byte)(_enabled ? FunctionSet.PowerOn : FunctionSet.PowerOff), (byte)DisplayControl.InverseVideoMode } : new byte[] { (byte)FunctionSet.PowerOn, (byte)DisplayControl.NormalMode });
+            }
+        }
+
+        /// <summary>
+        /// Get or set the contrast from 0 to 127
+        /// </summary>
+        public byte Contrast
+        {
+            get => _contrast;
+            set
+            {
+                _contrast = value <= 127 ? value : throw new ArgumentException($"Contrast can only be between 0 and 127");
+                SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn | FunctionSet.ExtendedMode : FunctionSet.PowerOff | FunctionSet.ExtendedMode), (byte)(0x80 | _contrast) });
+            }
+        }
+
+        /// <summary>
+        /// Get or set the temperature coefficient
+        /// </summary>
+        public ScreenTemperature Temperature
+        {
+            get => _temperature;
+            set => SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn | FunctionSet.ExtendedMode : FunctionSet.PowerOff | FunctionSet.ExtendedMode), (byte)value });
+        }
+
+        #endregion
+
+        #region Primitive methods
+
+        /// <summary>
+        /// Draw what's in memory to the the screen
+        /// </summary>
+        public void Draw() => SpiWrite(true, _byteMap);
+
+        /// <summary>
+        /// Clear the screen
+        /// </summary>
+        public void Clear()
+        {
+            for (int i = 0; i < _byteMap.Length; i++)
+            {
+                _byteMap[i] = 0;
+            }
+
+            SetCursorPosition(0, 0);
+            Draw();
+        }
+
+        /// <summary>
+        /// Set the byte map
+        /// </summary>
+        /// <param name="byteMap">A 504 sized byte representing the full image</param>
+        public void SetByteMap(ReadOnlySpan<byte> byteMap)
+        {
+            if (byteMap.Length != ScreenBufferByteSize)
+            {
+                throw new ArgumentException($"{nameof(byteMap)} length have to be {ScreenBufferByteSize} bytes");
+            }
+
+            SetCursorPosition(0, 0);
+            byteMap.CopyTo(_byteMap);
+        }
+
+        #endregion
+
+        #region Text
+
+        /// <summary>
+        /// Write text
+        /// </summary>
+        /// <param name="text">The text to write</param>
+        public void Write(string text)
+        {
+            Span<byte> letter = stackalloc byte[6];
+            foreach (char c in text)
+            {
+                // We only display specific characters and ignore the rest
+                // And only if it's in the screen
+                if (_position <= ScreenBufferByteSize - 5)
+                {
+                    if (c >= 0x20 && c <= 0x7F)
```

Comment:
consider throwing when they're out of range. also for some characters like backspace you might consider moving back 1

Reply:
it's ok if it writes out of the screen, it's like for drawing. But great catch for backspace!